pkg/hubble/metrics/http/handler.go

```diff
@@ -69,50 +74,106 @@ func (h *httpHandler) Status() string {
 	if h.context == nil {
 		return ""
 	}
-	return h.context.Status()
+	return h.context.Status() + fmt.Sprintf(",exemplars=%t", h.exemplars)
 }
 
 func (h *httpHandler) ProcessFlow(ctx context.Context, flow *flowpb.Flow) error {
-	l7 := flow.GetL7()
+	if h.useV2 {
+		return h.processMetricsV2(flow)
+	} else {
+		return h.processMetricsV1(flow)
+	}
+}
+
+func (h *httpHandler) getMetadataFromFlow(flow *flowpb.Flow) (l7 *flowpb.Layer7, http *flowpb.HTTP, reporter string, traceID string, ok bool) {
+	l7 = flow.GetL7()
 	if l7 == nil {
-		return nil
+		return
 	}
-	http := l7.GetHttp()
+	http = l7.GetHttp()
 	if http == nil {
-		return nil
+		return
+	}
+	if l7.GetType() != flowpb.L7FlowType_REQUEST &&
+		l7.GetType() != flowpb.L7FlowType_RESPONSE {
+		return
 	}
 
-	reporter := "unknown"
+	reporter = "unknown"
 	switch flow.GetTrafficDirection() {
 	case flowpb.TrafficDirection_EGRESS:
 		reporter = "client"
 	case flowpb.TrafficDirection_INGRESS:
 		reporter = "server"
 	}
-	if h.useV2 {
-		if l7.Type != flowpb.L7FlowType_RESPONSE {
-			return nil
-		}
-		labelValues, err := h.context.GetLabelValuesInvertSourceDestination(flow)
-		if err != nil {
-			return err
-		}
-		status := strconv.Itoa(int(http.Code))
-		h.requests.WithLabelValues(append(labelValues, http.Method, http.Protocol, status, reporter)...).Inc()
-		h.duration.WithLabelValues(append(labelValues, http.Method, reporter)...).Observe(float64(l7.LatencyNs) / float64(time.Second))
-	} else {
-		labelValues, err := h.context.GetLabelValues(flow)
-		if err != nil {
-			return err
-		}
-		switch l7.Type {
-		case flowpb.L7FlowType_REQUEST:
-			h.requests.WithLabelValues(append(labelValues, http.Method, http.Protocol, reporter)...).Inc()
-		case flowpb.L7FlowType_RESPONSE:
-			status := strconv.Itoa(int(http.Code))
-			h.responses.WithLabelValues(append(labelValues, status, http.Method, reporter)...).Inc()
-			h.duration.WithLabelValues(append(labelValues, http.Method, reporter)...).Observe(float64(l7.LatencyNs) / float64(time.Second))
-		}
+
+	if h.exemplars {
+		traceID = flow.GetTraceContext().GetParent().GetTraceId()
+	}
+
+	ok = true
```

Comment:
We set `ok` to `true` even for the case where `traceID == ""`. Is this what we want? What is actually the meaning of `ok`? It seems to me that we could have much simpler code by returning `reporter` and `traceID` only.

Reply:
I should have written a doc comment for this. It returns ok = true, if it's an HTTP flow.

Reply:
Refactored. PTAL.